src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -346,12 +375,20 @@ boolean requestsBundledTemplate() {
             return !StringUtils.isBlank(localTemplateName);
         }
 
+        boolean requestSnapshot() {
+            boolean snapshotName = name.equals("snapshot");
+            boolean snapshotTemplate =
+                    StringUtils.isBlank(template) && StringUtils.isBlank(localTemplateName);
+            boolean snapshotFeatures = duration == 0 && maxSize == 0 && maxAge == 0;
+            return snapshotName && snapshotTemplate && snapshotFeatures;
+        }
+
         boolean isValid() {
             boolean requestsCustomTemplate = requestsCustomTemplate();
             boolean requestsBundledTemplate = requestsBundledTemplate();
             boolean requestsEither = requestsCustomTemplate || requestsBundledTemplate;
             boolean requestsBoth = requestsCustomTemplate && requestsBundledTemplate;
-            return requestsEither && !requestsBoth;
+            return requestsEither && !requestsBoth && !requestSnapshot();
```

Comment:
I think this might be more understandable written as `requestSnapshot || (requestsEither && !requestsBoth /* ie requests exactly one */)`